### ISSUE TITLE
Wrong inference results

### ISSUE THREAD

**Issue by walzimmer** [kujason/avod#101]
@kujason @melfm 
The problem is that proposals are generated too high. That is why the detections are also too high:

![master_124_in-9900_000016_detections_too_high](https://user-images.githubusercontent.com/22540236/44018767-1aab09f2-9edd-11e8-8982-338203c5a4e3.png)

I am using grayscale images and duplicated the first channel three times to correspond to the algortihm.
Training was done on the KITTI dataset and now I am doing inference on our own data.

Furthermore our camera was mounted at 1.293m height (KITTI has a height of ~1.65).
Setting the height to -1.293 instead of +1.293 the anchors will be generated on the ground plane (I have used different colors for better visualization):

![master_130_in-9900_000012_anchors_correct_m1 293](https://user-images.githubusercontent.com/22540236/44023106-df54cbc4-9ee9-11e8-89da-1db4c0c3bb71.png)

BEV generation is also correct:

![master_131_in-9900_000012_anchors_in_bev](https://user-images.githubusercontent.com/22540236/44024813-ad0c3760-9eee-11e8-85b5-709a34abb478.png)

But using a height of -1.293 I am not able to generate mini batches, because there are no points in the range from -0.2 to 2.3. The method create_slice_filter in kitti_utils will filter out all points in the pointcloud.

What is currently working is to use:
 - a height of +1.293 for generating mini batches
 - a height of -1.293 to generate the anchor grid
 - and a height of 0 to generate proposals 

The result is that proposals lie on the ground plane (having a correct height) but are still wrong in the lateral and longitudinal position:

![master_129_in-9900_000012_anchors_wrong_lat_position](https://user-images.githubusercontent.com/22540236/44023997-56701630-9eec-11e8-862a-7bf8a1ab5425.png)

Does anyone know why the proposals are generated at the wrong position?
The projection matrix should be fine, since the ground truth labels are projected correctly in red.
My guess is that we use a different camera with a focal length of 1613px (6.05mm).
Training was performed on the KITTI training dataset using cameras with three different focal lengths where the average is about 2.305mm)

**Comment by kujason** [comment]
Not clear what the issue is, would recommend trying to visualize the scene and generated proposals in 3D to debug problems.
